lib/modules/device/DevicesController.ts

```diff
@@ -56,6 +57,9 @@ export class DevicesController {
             { path: "device-manager/:engineId/devices/:_id", verb: "put" },
           ],
         },
+        upsert: {
+          handler: this.upsert.bind(this),
+        },
```

Comment:
It looks like it is missing the http configuration

Reply:
oupsyy ty

Reply:
done